quickjs.c

```diff
@@ -10083,6 +10085,19 @@ void JS_SetOpaque(JSValue obj, void *opaque)
    JSObject *p;
     if (JS_VALUE_GET_TAG(obj) == JS_TAG_OBJECT) {
         p = JS_VALUE_GET_OBJ(obj);
+        if (p->class_id > JS_CLASS_INIT_COUNT)
+            p->u.opaque = opaque;
+        // TODO(saghul) report error?
```

Comment:
Alternative options:

- abort
- make it return int, and report failure
- take a ctx parameter and throw an internal error

Since this is a programming error that likely implies things work in weird ways I'd go with either this PR or abort.

Thoughts @bnoordhuis ?

Reply:
Another option is to change JSObject.u.cfunc like this:
```diff
--- a/quickjs.c
+++ b/quickjs.c
@@ -955,7 +955,8 @@ struct JSObject {
             JSVarRef **var_refs;
             JSObject *home_object; /* for 'super' access */
         } func;
-        struct { /* JS_CLASS_C_FUNCTION: 12/20 bytes */
+        struct { /* JS_CLASS_C_FUNCTION: 16/28 bytes */
+            void *opaque;
             JSContext *realm;
             JSCFunctionType c_function;
             uint8_t length;
```
Downside: grows JSObject by a word

Reply:
Yeah I didn't want to make jsobject later. Plus, we'd need to add it outside I guess? Since one could get ahold of the other class types and break some shit by setting the opaque...